io-developer/php-whois
# Using symfony/polyfill-intl-idn instead of true/php-punycode

| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes    
| Fixed tickets | no   
| License       | MIT



`true/php-punycode` has been archived. Use `symfony/polyfill-intl-idn` to convert IDN.

----

Changes:
- Install `symfony/polyfill-intl-idn`, and drop `true/php-punycode` in `composer.json`
- Use `Symfony\Polyfill\Intl\Idn\Idn` to encode and decode.
- Modify some class names and interface names.
- Different test environment should get correct dependency by `composer udpate`, not `composer install`.
